### PR TITLE
[DEST-1068] Ability to use Advanced Matching on External ID

### PR DESCRIPTION
**Pull request by briemcnally**
**What does this PR do?**
Adds functionality to support Advanced Matching on External ID for Facebook Pixel. Since `external_id` is not part of the Segment spec I will introduce a new setting in Partner Portal marked for Client Side only called `keyForExternalId`. Customers can indicated the key in their traits which they would like to map to FB Pixel `external_id`. 

**Are there breaking changes in this PR?**
No breaking changes. This has been tested and validated with unit tests and locally using the Golden Compiler in my browser. 

**Any background context you want to provide?**
JIRA Ticket: https://segment.atlassian.net/browse/DEST-1068

Customer Commitment: October 1st

**Is there parity with the server-side/android/iOS integration components (if applicable)?**
We are not currently building support for this SS or on mobile. 

**Does this require a new integration setting? If so, please explain how the new setting works**
Yes we will be adding a new setting in PP. See the above comment in the description. 

**Links to helpful docs and other external resources**
N/A

### CODE REVIEW

File: integrations/facebook-pixel/lib/index.js [hunk 1]
```diff
@@ -582,6 +583,10 @@ FacebookPixel.prototype.formatTraits = function formatTraits(analytics) {
       .toLowerCase();
   var state = address.state && address.state.toLowerCase();
   var postalCode = address.postalCode;
+  var external_id; // eslint-disable-line
+  if (this.options.keyForExternalId.length) {
```
Comment (mericsson) [inline integrations/facebook-pixel/lib/index.js]: Interesting use of `.length`. `keyForExternalId` is guaranteed to be a string? I think so.

To be safer can do:

I think you can actually just do `if (this.options.keyForExternalId) {`

Should be equivalent to what you have.

File: integrations/facebook-pixel/test/index.test.js [hunk 2]
```diff
@@ -2554,6 +2554,45 @@ describe('Facebook Pixel', function() {
...
+        ge: 'm',
+        db: '19910113',
+        external_id: '123456789' // eslint-disable-line
+      };
+      var actual = facebookPixel.formatTraits(analytics);
+
+      analytics.deepEqual(expected, actual);
+    });
+
+    it('should reject an external_id if NOT defined in settings', function() {
```
Comment (mericsson) [inline integrations/facebook-pixel/test/index.test.js]: An additional test worthwhile to add is one where you do not specify `facebookPixel.options`.

File: integrations/facebook-pixel/lib/index.js (hunk 1 above)
Comment (briemcnally) [reply integrations/facebook-pixel/lib/index.js]: @mericsson Yes that was a typo, just pushed a commit to resolve.

File: integrations/facebook-pixel/test/index.test.js (hunk 2 above)
Comment (briemcnally) [reply integrations/facebook-pixel/test/index.test.js]: Just added

**Pull request merged**